internal/driver/glfw/canvas_test.go

```diff
@@ -204,6 +204,27 @@ func TestGlCanvas_Focus(t *testing.T) {
 	assert.True(t, o2e.focused)
 }
 
+func TestGlCanvas_Focus_BeforeVisible(t *testing.T) {
+	w := createWindow("Test")
+	w.SetPadded(false)
+	e := widget.NewEntry()
+	c := w.Canvas().(*glCanvas)
+	c.Focus(e) // this crashed in the past
+}
+
+func TestGlCanvas_Focus_SetContent(t *testing.T) {
+	w := createWindow("Test")
+	w.SetPadded(false)
+	e := widget.NewEntry()
+	w.SetContent(e)
+	c := w.Canvas().(*glCanvas)
+	c.Focus(e)
+	assert.Equal(t, e, c.Focused())
+
+	w.SetContent(e)
```

Comment:
Because this test will be false positive as soon as someone optimises `SetContent` with `if c == canvas.content { return }`, I propose the following:

```suggestion
	w.SetContent(container.NewHBox(e))
	c := w.Canvas().(*glCanvas)
	c.Focus(e)
	assert.Equal(t, e, c.Focused())

	w.SetContent(container.NewVBox(e))
```

Also, it would be helpful to have a test which checks that `SetContent` resets the focus if the current focused object is not part of the new content.